source/engine/graphics/api/vulkan/RenderGraph.cpp

```diff
@@ -229,15 +227,14 @@ void RenderGraph::BuildRenderGraph(const PerFrameRenderStateData& stateData, uin
         renderGraph.isSinkDirty.at(RenderPassSinkType::ShadowMap) = true;
         m_sinkBuffers.at(RenderPassSinkType::ShadowMap).Clear();
 
-        renderGraph.shadowPasses.clear();
-        for (auto i : std::views::iota(0u, stateData.omniDirLightsCount))
+        while (renderGraph.shadowPasses.size() < stateData.omniDirLightsCount + stateData.uniDirLightsCount)
         {
-            renderGraph.shadowPasses.push_back(CreateShadowMappingPass(m_device, m_gpuAllocator, m_swapchain, m_shaderBindingManager, m_dimensions, i, true));
+            renderGraph.shadowPasses.push_back(CreateShadowMappingPass(m_device, m_gpuAllocator, m_swapchain, m_shaderBindingManager, m_dimensions));
         }
 
-        for (auto i : std::views::iota(stateData.omniDirLightsCount, stateData.omniDirLightsCount + stateData.uniDirLightsCount))
+        while (renderGraph.shadowPasses.size() > stateData.omniDirLightsCount + stateData.uniDirLightsCount)
```

Comment:
These while loops seem weird to me. Won't only one of them run, depending on whether the new size is greater or less than the current? To me, that'd be better expressed with an if statement. You could also do reserve + generate_n when we need to add, and resize when we need to shrink.

Reply:
Yes, exactly. I like the reserve/generate_n and resize approach